src/main/java/codechicken/nei/ItemPanel.java

```diff
@@ -1,85 +1,83 @@
 package codechicken.nei;
 
-import codechicken.lib.vec.Rectangle4i;
 import codechicken.nei.api.GuiInfo;
 import codechicken.nei.api.INEIGuiHandler;
 import codechicken.nei.guihook.GuiContainerManager;
+import codechicken.nei.ItemsGrid.ItemPanelSlot;
 import codechicken.nei.recipe.GuiCraftingRecipe;
 import codechicken.nei.recipe.GuiRecipe;
 import codechicken.nei.recipe.GuiUsageRecipe;
+import codechicken.nei.recipe.StackInfo;
 import net.minecraft.client.gui.inventory.GuiContainer;
+import net.minecraft.client.gui.inventory.GuiContainerCreative;
 import net.minecraft.inventory.Slot;
 import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.IFluidContainerItem;
 
 import java.util.ArrayList;
 
-import static codechicken.lib.gui.GuiDraw.drawRect;
-import static codechicken.nei.NEIClientConfig.canPerformAction;
-
-public class ItemPanel extends Widget {
-    /**
-     *  Backwards compat :-/
-     */
-    public static ArrayList<ItemStack> items = new ArrayList<>();
```

Comment:
Please don't remove public variables.  Other mods tend to rely on them and then crash when they're not present.  Feel free to mark them as @deprecated, but they need to stay in.

Reply:
The variable was never filled.

![image](https://user-images.githubusercontent.com/31038811/147885548-fd7aec1d-b018-45cf-885a-ee3822f752ac.png)

If i save it, it always was be empty. it's fine?

Reply:
That's fine, at least the other mods that depend on it wont' crash :)

Mark it with @deprecated